test/dlc_tests/conftest.py

```diff
@@ -176,17 +185,19 @@ def ec2_instance(
                 LOGGER.error(f"Failed to launch in {a_zone} with Error: {e}")
                 continue
     else:
-        instances = ec2_resource.create_instances(**params)
+        try:
+            instances = ec2_resource.create_instances(**params)
+        except ClientError as e:
+            if "InsufficientInstanceCapacity" in str(e):
+                LOGGER.warning(f"Failed to launch {ec2_instance_type} in {region} because of insufficient capacity")
+                if ec2_instance_type in ec2_utils.ICE_SKIP_INSTANCE_LIST:
+                    return None
```

Comment:
Is it possible to add the pytest skip directly here instead of returning None and expecting that to be handled by the callers?

Reply:
Haven't tried this. Will run a couple of tests to see if this works.